src/actions/icons.js

```diff
@@ -59,8 +63,17 @@ export function iconStartLoading(icon: string): ThunkAction<Promise<void>> {
           // nothing to do
           break;
         default:
-          throw new Error(`Unknown icon load result ${result}`);
+          throw new Error(`Unknown icon load result ${result.type}`);
       }
     });
   };
 }
+
+/**
+ * Transforms a URL into a valid CSS class name.
+ */
+async function _classNameFromUrl(url): Promise<string> {
+  return url.startsWith('data:image/')
+    ? 'dataUrl' + (await sha1(url))
+    : url.replace(/[/:.+>< ~()#,]/g, '_');
+}
```

Comment:
Also I believe you moved how the classname was generated here just becase of the fact you made it async. But I don't mind, I feel like it's better to generate the classnames once rather than regenerating them always. And it makes things easier with the counter.